<launch>

    <!-- Load the config file for the joint controller on the parameter server -->
    <rosparam command="load" file="$(find arduinobot_controller)/config/joints.yaml"  />

    <!-- Start the controllers  -->
    <node name="controller_spawner" pkg="controller_manager" type="spawner" respawn="false"
        output="screen" ns="/arduinobot_sim" args="joint_1_position_controller joint_2_position_controller joint_3_position_controller joint_4_position_controller joint_5_position_controller joint_state_controller"/>

    <!-- Start the controller interface that converts joint state messages into motor actions  -->
    <node pkg="arduinobot_controller" type="basic_fake_controller_interface.py" name="basic_fake_controller_interface" respawn="true" output="screen"/>

    <!-- Launch the joint state publisher GUI to control the joint angles -->
    <node name="joint_state_publisher_gui" pkg="joint_state_publisher_gui" type="joint_state_publisher_gui"/>
  
</launch>